homeassistant/components/thermostat/zwave.py

```diff
@@ -0,0 +1,108 @@
+"""
+ZWave Thermostat.
+
+"""
+
+# Because we do not compile openzwave on CI
+# pylint: disable=import-error
+from homeassistant.components.thermostat import DOMAIN
+from homeassistant.components.thermostat import ThermostatDevice
+from homeassistant.components.zwave import (
+    ATTR_NODE_ID, ATTR_VALUE_ID, NETWORK,
+    ZWaveDeviceEntity)
+from homeassistant.const import TEMP_FAHRENHEIT
+
+CONF_NAME = 'name'
+DEFAULT_NAME = 'ZWave Thermostat'
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the ZWave thermostats."""
+    if discovery_info is None or NETWORK is None:
+        return
+
+    node = NETWORK.nodes[discovery_info[ATTR_NODE_ID]]
+    value = node.values[discovery_info[ATTR_VALUE_ID]]
+    value.set_change_verified(False)
+    add_devices([ZWaveThermostat(value)])
+
+
+# pylint: disable=too-many-arguments
+class ZWaveThermostat(ZWaveDeviceEntity, ThermostatDevice):
+    """Represents a HeatControl thermostat."""
+    def __init__(self, value):
+        from openzwave.network import ZWaveNetwork
+        from pydispatch import dispatcher
+        from homeassistant.helpers.temperature import convert
```

Comment:
I have moved all the imports to the top in my test thermostat, and it
doesn't seem to affect the network at all.

On Sun, Mar 20, 2016 at 10:14 PM, Paulus Schoutsen <notifications@github.com
> wrote:

> In homeassistant/components/thermostat/zwave.py
> <https://github.com/balloob/home-assistant/pull/1416#discussion_r56772266>
> :
>
> > +    if discovery_info is None or NETWORK is None:
> > +        return
> > +
> > +    node = NETWORK.nodes[discovery_info[ATTR_NODE_ID]]
> > +    value = node.values[discovery_info[ATTR_VALUE_ID]]
> > +    value.set_change_verified(False)
> > +    add_devices([ZWaveThermostat(value)])
> > +
> > +
> > +# pylint: disable=too-many-arguments
> > +class ZWaveThermostat(ZWaveDeviceEntity, ThermostatDevice):
> > +    """Represents a HeatControl thermostat."""
> > +    def __init__(self, value):
> > +        from openzwave.network import ZWaveNetwork
> > +        from pydispatch import dispatcher
> > +        from homeassistant.helpers.temperature import convert
>
> I'm just talking about the homeassistant import. It got added here when
> you accepted the PR by @luxus <https://github.com/luxus>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/balloob/home-assistant/pull/1416/files/ad6da437e6e83261e049f9c669a042c72998aa9d#r56772266>
>